mod_reforged/hooks/contracts/contracts/return_item_contract.nut

```diff
@@ -0,0 +1,46 @@
+::mods_hookExactClass("contracts/contracts/return_item_contract", function(o) {
+    local createStates = o.createStates;
+    o.createStates = function()
+    {
+        createStates();
+        foreach (state in this.m.States)
+        {
+            if (state.ID != "Offer") continue;
```

Comment:
You _only_ want to change something in the `Offer` state. So, please do a `break` at the end of the loop, or enclose this in a positive `if (state.ID == "Offer")` and `break` it inside that if block. In my opinion, when you have a single condition, it is better (for readability) to do a positive check than an extracted inverse check like this. Extraction and inversion is useful when you have several checks.

Reply:
How will this case now work under this SOP point?:

> 13. Don't use guard clauses for only one level of indentation.

With the following for-loop we now have two levels of indentation which this guard clause helps to minimize back to one.
I feel like this is exactly the case that guard clauses under `13.` is for now. Keeping nested indentations to a minimum

Reply:
Guards to me feel like "Run the code for all situations except for the condition in the guard". But what you want here is "Run the code for this one particular condition". It just trips me when I look at this. E.g. I'd be ok with a guard here if the guard was `if (state.ID == "Offer") continue`